compiler/rustc_typeck/src/check/cast.rs

```diff
@@ -79,6 +79,21 @@ enum PointerKind<'tcx> {
     OfParam(&'tcx ty::ParamTy),
 }
 
+impl PointerKind<'_> {
+    fn maybe_wide(&self) -> bool {
+        *self != PointerKind::Thin
+    }
+
+    /// The name of the metadata that the wide pointer contains
+    fn known_metadata(&self) -> Option<&'static str> {
+        match self {
+            PointerKind::Length => Some("a length"),
+            PointerKind::Vtable(..) => Some("a vtable"),
+            _ => None,
+        }
+    }
```

Comment:
I'd inline and remove both of these functions, `maybe_wide` is trivial, and `known_metadata` is used only once and is unlikely to be useful anywhere else.